adr/ADR-47.md

```diff
@@ -0,0 +1,123 @@
+* Optional
+* Language specific implementation
+* If sentinel is supplied and total timeout is not set, total timeout defaults to the connection or system timeout.
+
+## Notes
+
+### Message Handling
+
+Each client must determine how to give messages to the user.
+* They could all be collected and given at once.
+* They could be put in an iterator, queue, channel, etc.
+* A callback could be made.
+
+### End of Data
+
+The developer should notify the user when the request has stopped processing and the receiving mechanism is not fixed like a list
+or iterator that termination is obvious. A queue or a callback for instance, should get a termination message.
+Implementation is language specific based on control flow.
+
+### Status Messages / Server Errors
+
+If a status (like a 503) or an error comes in place of a user message, this is terminal.
+This is probably useful information for the user and can be conveyed as part of the end of data.
+
+#### Callback timing
+
+If callbacks are made in a blocking fashion, 
+the client must account for the time it takes for the user to process the message 
+and not consider that time against the timeouts.
+
+### Sentinel
+
+If the client supports a sentinel with a callback/predicate that accepts the message and returns a boolean, 
+a return of true would mean continue to process and false would mean stop processing.
+
+If possible, the client should support the "standard sentinel", which is a message with a null/nil or empty payload.
+
+### Cancelling
+
+If possible, the user should be able to cancel the request. This is another pathway besides sentinel
+allowing that the dev can cancel the entire request-many arbitrarily.
+
+## Disconnection
+
+It's possible that there is a connectivity issue that prevents messages from reaching the requester,
+It might be difficult to differentiate that timeout from a total or stall timeout. 
+If possible to know the difference, this could be conveyed as part of the end of data. 
+
+## Strategies
+It's acceptable to make "strategies" via enum / api / helpers / builders / whatever.
+Strategies are just pre-canned configurations, for example:
+
+**Timeout or Wait** - this is the default strategy where only the total timeout is used.
+
+**Stall** - the stall defaults to the lessor of 1/10th of the total wait time (if provided) or the default connection timeout.
+
+**Max Responses** - accepts a max response number and uses the default timeout.
+
+### Max Responses Optimization
+On requests that specify max responses, and when not using mux inboxes, the client can unsubscribe with a count immediately after subscribing.
+Theoretically this unsub could be processed after a reply has come in and out of the server, so you still must check the count manually.
+
+### Best Practice
+Since the client is in charge of the subscription, it should always unsubscribe upon completion of the request handling instead of leaving it up to the server to time it out.
```

Comment:
would only happen when not using the mux version/new style, if using old style/non mux it must always clean up to avoid leaking subscriptions, that is a very common issue when users implement their own version of request/response expecting multiple responses